src/mem/cache/prefetch/bop.cc

```diff
@@ -234,21 +247,21 @@ BOP::calculatePrefetch(const PrefetchInfo &pfi,
     Addr tag_x = tag(addr);
 
     if (delayQueueEnabled) {
-        insertIntoDelayQueue(tag_x);
+        insertIntoDelayQueue(addr);
     } else {
-        insertIntoRR(tag_x, RRWay::Left);
+        insertIntoRR(addr, tag_x, RRWay::Left);
     }
 
     // Go through the nth offset and update the score, the best score and the
     // current best offset if a better one is found
-    bestOffsetLearning(tag_x);
+    bestOffsetLearning(addr);
 
-    // This prefetcher is a degree 1 prefetch, so it will only generate one
-    // prefetch at most per access
     if (issuePrefetchRequests) {
-        Addr prefetch_addr = addr + (bestOffset << lBlkSize);
-        addresses.push_back(AddrPriority(prefetch_addr, 0));
-        DPRINTF(HWPrefetch, "Generated prefetch %#lx\n", prefetch_addr);
+        for (int d = 1; d <= degree; d++) {
```

Comment:
Nit: in the paper he uses d as offset. I suggest renaming this variable to avoid confusing

Reply:
Fixed in the latest commit